### PR TITLE
Add `commit_sha`

### PR DESCRIPTION
**Pull request by nitrocode**
## what
* Add `commit_sha`

## why
* Pin environments to a version other than the default branch.

e.g.

```yaml
components:
  terraform:
    service-pr-123:
      component: preview
      settings:
        spacelift:
          repository: service
          branch: branch-of-pr-123-or-whatever-it-is
          # The path of the terraform directory in this repo should evaluate to "terraform/preview"
          # due to the base component name defined above.
          # NOTE: it looks like "custom_component_folder" usage has been replaced by the "base_component" value
          components_path: terraform
          # components_path: .
          commit_sha: 1c0d07776373a614355b18ada6da6897476c029d
          spacelift_run_enabled: true
        vars:
          enabled: true
          name: service-pr-123
          # name: service
          # attributes: ["pr-123"]
```

## references
* https://registry.terraform.io/providers/spacelift-io/spacelift/latest/docs/resources/run

### CODE REVIEW

**Review by osterman — APPROVED** [review]

**Comment by nitrocode** [comment]
I'm waiting to hear back from @aknysh before merging. I'm unsure on how to correctly test this out locally.

**Comment by aknysh** [comment]
> I'm waiting to hear back from @aknysh before merging. I'm unsure on how to correctly test this out locally.

No way to test it locally, I'll approve it and let's merge and test on a real account

**Review by aknysh — APPROVED** [review]

File: modules/stack/main.tf [hunk 1]
```diff
@@ -21,6 +21,24 @@ resource "spacelift_stack" "default" {
   terraform_workspace = var.terraform_workspace
 }
 
+resource "spacelift_run" "this" {
+  count = var.enabled && var.commit_sha != null ? 1 : 0
```
Comment (nitrocode) [inline modules/stack/main.tf:25]: If we do not want the `spacelift_run` resource based on the `var.commit_sha`, then we can make it based on a new boolean called `var.spacelift_run`.

```suggestion
  count = var.enabled && var.spacelift_run ? 1 : 0
```

File: modules/stack/main.tf [hunk 2]
```diff
@@ -21,6 +21,24 @@ resource "spacelift_stack" "default" {
...
+
+  stack_id   = spacelift_stack.default[0].id
+  commit_sha = var.commit_sha
+
+  keepers = {
+    branch = spacelift_stack.default[0].branch
+  }
```
Comment (nitrocode) [inline modules/stack/main.tf:32]: The branch will likely not change so we could remove these keepers. We do not need a keeper on the `commit_sha` itself.

**Review by aknysh — APPROVED** [review]

**Pull request merged**